test/ode/signals/test_signals.py

```diff
@@ -17,213 +17,943 @@
+        self.assertAllClose(self.sig_sum2(t), self.signal2(t) - self.signal3(t))
+        self.assertAllClose(self.double_sig_sum(t), self.signal1(t) + self.signal3(t))
+        t = 1.23
+        self.assertAllClose(self.sig_sum1(t), self.signal1(t) + self.signal2(t))
+        self.assertAllClose(self.sig_sum2(t), self.signal2(t) - self.signal3(t))
+        self.assertAllClose(self.double_sig_sum(t), self.signal1(t) + self.signal3(t))
+
+    def test_call_vectorized(self):
+        """Test vectorized __call__."""
+        t_vals = np.array([0.0, 1.23])
+        self.assertAllClose(
+            self.sig_sum1(t_vals), [self.signal1(t) + self.signal2(t) for t in t_vals]
+        )
+        self.assertAllClose(
+            self.sig_sum2(t_vals), [self.signal2(t) - self.signal3(t) for t in t_vals]
+        )
+        self.assertAllClose(
+            self.double_sig_sum(t_vals), [self.signal1(t) + self.signal3(t) for t in t_vals]
+        )
+        t_vals = np.array([[0.0, 1.23], [0.1, 2.0]])
+        self.assertAllClose(
+            self.sig_sum1(t_vals),
+            [[self.signal1(t) + self.signal2(t) for t in t_row] for t_row in t_vals],
+        )
+        self.assertAllClose(
+            self.sig_sum2(t_vals),
+            [[self.signal2(t) - self.signal3(t) for t in t_row] for t_row in t_vals],
+        )
+        self.assertAllClose(
+            self.double_sig_sum(t_vals),
+            [[self.signal1(t) + self.signal3(t) for t in t_row] for t_row in t_vals],
+        )
+
+    def test_conjugate(self):
+        """Verify conjugate() functioning correctly."""
+
+        sig_sum1_conj = self.sig_sum1.conjugate()
+        self.assertAllClose(
+            sig_sum1_conj.complex_value(2.313), np.conjugate(self.sig_sum1.complex_value(2.313))
+        )
+        self.assertAllClose(
+            sig_sum1_conj.complex_value(0.1232), np.conjugate(self.sig_sum1.complex_value(0.1232))
+        )
+
+
+class TestDiscreteSignalSum(TestSignalSum):
+    """Tests for DiscreteSignalSum."""
+
+    def setUp(self):
+        self.signal1 = Signal(np.vectorize(lambda t: 0.25), carrier_freq=0.3)
+        self.signal2 = Signal(lambda t: 2.0 * (t ** 2), carrier_freq=0.1)
+        self.signal3 = Signal(lambda t: 2.0 * (t ** 2) + 1j * t, carrier_freq=0.1, phase=-0.1)
+
+        self.sig_sum1 = DiscreteSignalSum.from_SignalSum(
+            self.signal1 + self.signal2, dt=0.5, start_time=0, n_samples=10
+        )
+        self.sig_sum2 = DiscreteSignalSum.from_SignalSum(
+            self.signal2 - self.signal3, dt=0.5, start_time=0, n_samples=10
+        )
+
+        self.signal1 = DiscreteSignal.from_Signal(self.signal1, dt=0.5, start_time=0, n_samples=10)
+        self.signal2 = DiscreteSignal.from_Signal(self.signal2, dt=0.5, start_time=0, n_samples=10)
+        self.signal3 = DiscreteSignal.from_Signal(self.signal3, dt=0.5, start_time=0, n_samples=10)
+
+        self.double_sig_sum = self.sig_sum1 - self.sig_sum2
```

Comment:
Is this used anywhere?

Reply:
Yeah - it inherits from `TestSignalSum`, so all of the test cases from that class are being run on these signals.